# origin: Debian (joshk)
# cset: n/a
# inclusion: wtf?
# revision date: 2005-02-05

Just a diff version of the ChangeLog..

--- /dev/null	2005-02-04 00:03:50.000000000 -0800
+++ a/ChangeLog	2005-02-05 16:32:55.000000000 -0800
@@ -0,0 +1,575 @@
+final:
+
+- 2.4.29-rc4 was released as 2.4.29 with no changes.
+
+
+Summary of changes from v2.4.29-rc3 to v2.4.29-rc4
+============================================
+
+Marcelo Tosatti:
+  o Cset exclude: solar@openwall.com|ChangeSet|20041218011100|24870
+  o Changed VERSION to 2.4.29-rc4
+  o Update i386 defconfig
+
+
+Summary of changes from v2.4.29-rc2 to v2.4.29-rc3
+============================================
+
+<raivis:mt.lv>:
+  o [NEIGH]: Calculate hash_val after possible table growth, not before
+
+<scott:sonic.net>:
+  o Fix net neighbour hash bug
+
+David S. Miller:
+  o [TG3]: Return 0 when PHY read times out, not all-ones
+  o [TG3]: Fix signedness issues in PHY read/write loops
+  o [TG3]: Update driver version and reldate
+  o [INET_ECN]: Add INET_ECN_* enumeration from 2.6.x
+
+Hideaki Yoshifuji:
+  o [IPV6]: Fix tunnel list locking in sit.c
+
+Jamal Hadi Salim:
+  o [NET]: Add ETH_P_MPLS_* and ARPHRD_INFINIBAND defines from 2.6.x
+
+Marcelo Tosatti:
+  o Adrian/Arjan/Marcelo: change tty_wakeup/tty_ldisc_flush to non-GPL export for compat reasons and change requirement to modutils 2.4.10
+  o Changed VERSION to 2.4.29-rc3
+
+Olaf Kirch:
+  o [NET]: Fix CMSG32_OK macros
+
+Patrick McHardy:
+  o [NETFILTER]: Associate locally generated ICMP errors with conntrack of original packet
+  o [NETFILTER]: Remove CONFIG_IP_NF_NAT_LOCAL config option
+  o [NETFILTER]: Save a level of indentation in icmp_reply_translation
+  o [NETFILTER]: Apply PRE_ROUTING manips in LOCAL_OUT for locally generated icmp errors
+  o [NETFILTER]: Verify NAT manips have been applied before reversing them in icmp_reply_translation
+  o [NETFILTER]: Release dst_entry in PRE_ROUTING after NAT
+  o [NETFILTER]: Fix stack leakage in iptables/ip6_tables
+
+Phil Oester:
+  o [NETFILTER]: revert MASQUERADE optimization for mostly static IPs
+
+Sridhar Samudrala:
+  o [SCTP] Fix potential null pointer dereference in sctp_err_lookup()
+  o [SCTP] Code cleanup: remove unused code and make needlessly global code static
+  o [SCTP] Treat ICMP protocol unreachable errors from non-SCTP capable hosts as ABORTs.
+  o [SCTP] Validate and respond to invalid chunk/parameter lengths
+  o [SCTP] Implementation of SCTP Implementer's Guide Section 2.35
+  o [SCTP] Clean up the T3_rtx timer when deleting a transport
+  o [SCTP] Fix bug in setting ephemeral port in the bind address
+  o [SCTP] Fix misc. issues in SCTP_PEER_ADDR_PARAMS set socket option
+  o [SCTP] Remove sk_xxx macros to be consistent with the rest of networking code and to avoid backporting issues.
+  o [SCTP] Fix sctp_getladdrs() to return valid local addresses on an endpoint that is bound to INADDR_ANY or inaddr6_any.
+
+Thomas Graf:
+  o [PKT_SCHED]: dsmark should ignore ECN bits
+
+Wensong Zhang:
+  o [IPVS]: change to run master/backup sync daemon at a time
+
+Yasuyuki Kozakai:
+  o [NETFILTER]: Backport fixes for ip6t_LOG
+  o [NETFILTER]: Backport fixes for ip6t_dst
+  o [NETFILTER]: Fix check for ESP header size in ip6t_esp
+  o [NETFILTER]: Backport fixes for ip6t_eui64
+  o [NETFILTER]: Backport fixes for ip6t_frag
+  o [NETFILTER]: Backport fixes for ip6t_hbh
+  o [NETFILTER]: Backport fixes for ip6t_ipv6header
+  o [NETFILTER]: Backport fixes for ip6t_multiport
+  o [NETFILTER]: Backport fixes for ip6t_rt
+  o [NETFILTER]: Backport fixes for ip6tables
+
+
+
+
+Summary of changes from v2.4.29-rc1 to v2.4.29-rc2
+============================================
+
+Christoph Hellwig:
+  o [XFS] make sure to always reclaim inodes in xfs_finish_reclaim
+  o [XFS] Fix NFS inode data corruption
+  o [XFS] Disable variable sized transfers on loop devices
+  o [XFS] Fix compilations for parisc
+
+Geert Uytterhoeven:
+  o Kill unused variables in the tty code
+  o Kill unused variables in the net code
+
+Jan Harkes:
+  o Fix Coda bugs found by Coverity checker
+
+Marcelo Tosatti:
+  o Update Dave Jones email address in MAINTAINERS file
+  o Linus Torvalds: Warn if mmap_sem is not locked in do_brk
+  o Change do_uselib() fix to match v2.6, rip do_brk_locked()
+  o Brad Spengler: Fix random poolsize sysctl (from 2.6.10-ac)
+  o Alan Cox: Fix moxa serial bound checking issue (from 2.6.10-ac)
+  o Brad Spengler: Fix RLIMIT_MEMLOCK issue
+  o get_user_pages: Change BUG_ON to WARN_ON
+  o Alan Cox: rose_rt_ioctl lack of bounds checking, reported by Coverity (from 2.6.10-ac)
+  o Alan Cox: sdla_xfer lack of bounds checking, reported by Coverity (from 2.6.10-ac)
+  o Makefile
+  o Revert dubious get_user_pages() bug checking
+  o Olaf Kirch: sendmsg compat wrapper fixes
+  o Cset exclude: marcelo@logos.cnet|ChangeSet|20050110190211|08215
+  o Fix expand_stack() SMP race
+  o Add missing Documentation/tty.txt from tty/ldisc locking updates
+  o Completly remove old do_brk() fix
+  o Linus Torvalds: Create helper for mmap_sem write-lock check in do_brk()
+  o Fix mmap.c typo
+
+Mikael Pettersson:
+  o sungem UniNorth 2 GMAC support
+
+Nathan Scott:
+  o [XFS] Add sanity checks before use of attr_multi opcount parameter
+
+Pete Zaitcev:
+  o EHCI race fix
+
+
+Summary of changes from v2.4.29-pre3 to v2.4.29-rc1
+============================================
+
+<a.pugachev:pcs-net.net>:
+  o drivers/net/appletalk/Config.in depends on CONFIG_ATALK
+
+<albertcc:tw.ibm.com>:
+  o [libata] use PIO mode for request sense
+  o [libata] PIO error handling improvement
+  o [libata] verify ATAPI DMA for a given request is OK
+
+<gortan:tttech.com>:
+  o 8139cp: support for TTTech MC322
+
+<jason.d.gaston:intel.com>:
+  o SATA support for Intel ICH7
+
+<mbellon:mvista.com>:
+  o 32 bit ltrace oops when tracing 64 bit executable [X86_64]
+
+<mhw:wittsend.com>:
+  o Computone driver update
+
+<penguin:muskoka.com>:
+  o 8390 Tx fix for non i386
+
+<stkn:gentoo.org>:
+  o [libata] add #include (fixes 2.4 alpha build)
+
+Don Fry:
+  o pcnet32: added pci_disable_device
+
+Ganesh Venkatesan:
+  o e1000: ITR does not default correctly on 2.4.x kernels
+  o e1000: Fix for kernel panic when the interface is brought down while the NAPI enabled driver is under stress
+  o e1000: Fix ethtool diagnostics -- specifically for blade server implementations
+  o e1000: Enabling NETIF_F_SG without checksum offload is illegal
+  o e1000: remove a redundant assignment to a local nr_frags in e1000_xmit_frame
+  o e1000: Synchronizing multicast table setup with transmit path - ak@suse.de
+  o e1000: fix tx resource cleanup logic
+  o e1000: {set, get}_wol is now symmetric for 82545EM adapters
+  o e1000: Sparse cleanup - shemminger@osdl.org
+  o e1000: Added workaround to prevent inadvertent gigabit waveform to be sent out on the wire due to init-time operations on the IGP phy.
+  o e1000:  Applied eeprom fix where it was possible to read/write
+  o e1000:  Applied smart speed fix where the code was forcing smart speed on all the time.  Now it will honor the setting defined in the eeprom.
+  o e1000: Driver version number, white spaces, comments, device id & other changes
+  o e1000: Documentation/networking/e1000.txt update
+
+Ian Kent:
+  o autofs4 add missing compat ioctls
+
+Jean Delvare:
+  o I2C: Cleanup a couple media/video drivers
+
+Jeff Garzik:
+  o [libata sata_nv] fix dev detect by removing sata-reset flag
+  o [libata sata_uli] add 5281 support, fix SATA phy setup for others
+
+Marcelo Tosatti:
+  o Changed VERSION to 2.4.29-rc1
+  o Paul Starzetz: sys_uselib() race vulnerability  (CAN-2004-1235)
+
+Margit Schubert-While:
+  o prism54 sync with 2.6
+  o prism54 fix resume processing
+  o prism54 sparse fixes
+
+Paul Mackerras:
+  o PPC64 signal code cleanup
+
+Pete Zaitcev:
+  o USB: Add user defined IDs to ftdi
+
+Ralf Bächle:
+  o MIPS network drivers
+  o NE2000 on Toshiba RBTX4927 fixes
+
+Solar Designer:
+  o Check for zero program header on load_elf_interp()
+
+
+
+
+Summary of changes from v2.4.29-pre2 to v2.4.29-pre3
+============================================
+
+<baris:idealteknoloji.com>:
+  o Remove msleep() definitions from sx8.c and forcedeth.c: it is generic now
+
+Andi Kleen:
+  o x86_64: fix signal restart bug
+  o [CAN-2004-1144] Fix int 0x80 hole in 2.4 x86-64 linux kernels
+
+Andries E. Brouwer:
+  o do not use CONFIG_BLK_STATS
+
+Chris Wright:
+  o a.out: error check on set_brk
+  o Backport of 2.6 fix to insert_vm_struct to make it return an error rather than BUG()
+
+David S. Miller:
+  o [SPARC]: Adjust 32-bit ELF_ET_DYN_BASE
+
+Geert Uytterhoeven:
+  o m68k: fix incorrect config comment in check_bugs()
+
+H. J. Lu:
+  o backport v2.6: Fix pty race condition
+
+Ian Abbott:
+  o serial closing_wait and close_delay used from wrong data structure
+
+Marcelo Tosatti:
+  o Solar Designer: Fix do_follow_link() comment
+  o Jason Baron: Backport v2.6 tty/ldisc locking fixes
+  o Move msleep() from libata-compat.h to generic headers
+  o Cset exclude: akpm@osdl.org|ChangeSet|20041218001750|00972
+  o Changed EXTRAVERSION to -pre3
+  o Cset exclude: trond.myklebust@fys.uio.no|ChangeSet|20040521160141|29598
+  o Fix NFS hang on unlink problems: cset exclude: trond.myklebust@fys.uio.no|ChangeSet|20041110174036|20706
+
+Simon Horman:
+  o binfmt_elf force_sig arguments fix
+
+Solar Designer:
+  o Fix booting off USB CD-ROMs (do_mounts.c)
+  o binfmt_elf fix return error codes and early corrupt binary detection
+  o procfs enhanced error reporting
+  o sysctl: block bogus argument earlier
+  o stronger (paranoic) mremap argument checking
+
+Stephen Hemminger:
+  o [TCP]: Missing newline character in printk
+  o [PKT_SCHED]: netem: restart device after inserting packets
+
+Thomas Graf:
+  o [PKT_SCHED]: Fix double locking in tcindex destroy path
+
+
+
+Summary of changes from v2.4.29-pre1 to v2.4.29-pre2
+============================================
+
+<alexn:dsv.su.se>:
+  o Correct /dev/mptctl major number in Configure.help
+
+<nboullis:debian.org>:
+  o [SPARC64]: Add SMB_IOC_GETMOUNTUID32 to compat ioctl table
+
+<tv:lio96.de>:
+  o VM documentation fix: vm_anon_lru default is 0
+
+Adrian Bunk:
+  o remove bouncing email address of Deanna Bonds
+  o add missing SCSI_SATA_AHCI Configure.help entry
+  o USB_ETH{,_RNDIS} EXPERIMENTAL dependencies
+  o let SCSI_SATA_NV depend on EXPERIMENTAL
+  o Update email address of Philip Blundell
+  o update email address of Andrea Arcangeli
+
+Cal Peake:
+  o remove obsolete PIIX config help
+
+Chris Wright:
+  o proc_tty.c warning fix
+  o [IPV4/IPV6]: IGMP source filter fixes
+
+Christoph Hellwig:
+  o [XFS] handle nfs requesting ino 0 gracefully
+  o [XFS] fix handling of bad inodes
+  o [XFS] remove useless S_ISREG check in ->mmap and ->mprotect
+  o [XFS] split pagebuf_get, use get/read_flags correctly
+  o [XFS] Fix declaration of _pagebuf_find to not be static
+  o [XFS] handle inode creation race
+  o [XFS] call the right function in pagebuf_readahead
+
+David Brownell:
+  o usb gadget updates: core
+  o usb gadget updates: ether/rndis
+  o usb gadget updates: Minor update to handle more hardware
+
+David S. Miller:
+  o [SPARC64]: Update defconfig
+  o [SPARC64]: Fix SMP cpu bringup bug when bigkernel
+  o [IPV4]: Do not leak IP options
+  o [NET]: CMSG compat code needs signedness fixes too
+
+Dean Roehrich:
+  o [XFS] Need to vn_revalidate after dm_set_fileattr
+  o [XFS] update a copyright notice
+
+Douglas Gilbert:
+  o off-by-1 libata-scsi INQUIRY VPD pages 0x80 and 0x83
+
+Eric Sandeen:
+  o [XFS] Wait for all async buffers to complete before tearing down the filesystem at umount time
+
+Geoffrey Wehrman:
+  o [XFS] Add xfs_rotorstep sysctl for controlling placement of extents for new files by the inode32 allocator.
+
+Herbert Xu:
+  o [NET]: Fix CMSG validation checks wrt. signedness
+
+Ingo Molnar:
+  o floppy boot-time detection fix
+
+Jakub Bogusz:
+  o don't recursively crash in die() on CHRP/PReP machines
+
+Jeff Garzik:
+  o [libata docs] add chapter on libata driver API
+  o [libata] only DMA map data for DMA commands (fix >=4GB bug)
+
+Jon Krueger:
+  o [XFS] Allow the option of skipping quotacheck processing
+
+Len Brown:
+  o [ACPI] acpi=off must disable acpi_early_init()
+
+Luca Tettamanti:
+  o radeonfb: don't try to ioreamp the entire VRAM
+  o Add new PCI id to radeonfb
+
+Marcelo Tosatti:
+  o Cset exclude: vince@arm.linux.org.uk|ChangeSet|20041125151649|65331
+  o backport v2.6 fork/thread file descriptor race fix
+  o p8022 unregister packet type on unload
+  o Make sure VC resizing fits in s16
+  o psnap correctly unregister on module exit
+  o Changed EXTRAVERSION to -pre2
+
+Nathan Scott:
+  o [XFS] Fix incorrect use of do_div on realtime device growfs code path
+  o [XFS] Fix some locking oddities in extended attributes code (ilock excl vs shared).
+  o [XFS] Convert to list_for_each_entry_safe form in reclaim list walk
+  o [XFS] Ensure bytes read statistic is not updated when the generic routines fail.
+  o [XFS] Add nosymlinks inode flag for the security folks, reserve projinherit flag.
+  o [XFS] Update XFS quota header - add macros, use standard gpl template
+  o [XFS] Make xfssyncd more generic, hand off out-of-space flushing to it; fixes two deadlocks when near-full and fixes a 4KSTACKS problem in XFS.
+  o [XFS] Remove crufty old cap/mac code - never used, never compiled, gone
+  o [XFS] Fix merge botch affecting xfs_setattr for realtime files
+  o [XFS] Simplify page probe/submit code so buffers bayond eof not dirtied/written.
+  o [XFS] Remove unused function prototypes
+
+Randy Dunlap:
+  o Fix unresolved symbol on x86-64: export swiotlb
+
+Solar Designer:
+  o [TCP]: Missing KERN_* in input path printks
+
+Stephen Hemminger:
+  o [UDP]: Select handling of bad checksums
+
+Timothy Shimmin:
+  o [XFS] xfs reservation issues with xlog_sync roundoff
+
+Wensong Zhang:
+  o [IPVS] add a sysctl variable to expire quiescent template
+
+
+Summary of changes from v2.4.28 to v2.4.29-pre1
+============================================
+
+<a.pugachev:pcs-net.net>:
+  o via82cxxx audio procfs code selection fix
+
+<edward_peng:alphanetworks.com>:
+  o dl2k: correct author's email
+
+<lnville:tuxdriver.com>:
+  o [VLAN]: change_mtu should return 0 on success
+
+<manfred99:gmx.ch>:
+  o [DECNET]: dn_neigh.c needs linux/module.h
+  o [ATM]: Force -n option in gzip invocation
+  o Tigran Aivazian: backport sigmatch() issue in microcode.c
+
+<mjagdis:eris-associates.co.uk>:
+  o Mike Jagdis CREDITS email address change
+
+<vince:arm.linux.org.uk>:
+  o vga16fb: Fix frame buffer bad memory mapping
+
+Adrian Bunk:
+  o remove outdated Stallion contact information
+
+Andrea Arcangeli:
+  o Lazily add anonymous pages to LRU
+
+Andries E. Brouwer:
+  o backport v2.6 largefile isofs fix
+
+Barry K. Nathan:
+  o Fix ELF exec with huge bss
+  o binfmt_elf.c fix for 32-bit apps with large bss
+
+Bartlomiej Zolnierkiewicz:
+  o REQUEST_SENSE support for ATAPI
+  o [libata] arbitrary size ATAPI PIO support
+  o arbitrary size ATAPI PIO support bugfixes
+  o make ATAPI PIO work
+  o libata PIO bugfix
+
+Chris Wright:
+  o /proc/tty/driver/serial reveals the exact number of characters used in serial links (CAN-2003-0461)
+
+Christoph Hellwig:
+  o fix sata_svw compile
+
+Dave Kleikamp:
+  o JFS: Fix extent overflow bugs
+  o JFS: avoid assert in lbmfree
+  o JFS: Fix endian errors
+  o JFS: fix race in jfs_commit_inode
+
+David S. Miller:
+  o [TCP]: Receive buffer moderation fixes
+  o [NETLINK]: sed 's/->sk_/->//' in af_netlink.c
+
+Ganesh Venkatesan:
+  o e1000: Update Documentation/networking/e1000.txt
+  o e1000: fix set_pauseparam for fiber serdes link
+  o e1000: remove unused function e1000_enable_mng_pass_thru
+  o e1000: fix set ringparam for ethtool returning error
+  o e1000: driver version update
+  o e1000: white space corrections
+  o e100: Update to Configure.help
+  o e1000: Update to Configure.help
+
+Herbert Xu:
+  o [NETLINK]: Backport pid hashing changes from 2.6
+  o [NETLINK]: Invoke netlink_proto_init() correctly in non-modular case
+
+Jan Kara:
+  o Configurable quota messages
+
+Jean Delvare:
+  o I2C updates for 2.4.28 (1/5)
+  o I2C updates: i2c proc parser fix
+  o I2C updates: hardcoded buffer size should depend on define
+  o I2C updates: lack trailing newline in logs
+  o I2C updates: get rid of unused code
+
+Jeff Garzik:
+  o [netdrvr dl2k] new TX scheme, fix minor bug
+  o [netdrvr dl2k] remove unused constant 'CFI'
+  o [libata] add AHCI driver
+  o [libata] fix minor 2.6 backport problems
+  o [libata] return ENOTTY rather than EOPNOTSUPP for unknown-ioctl
+  o [libata] use kunmap_atomic() correctly
+  o [libata] cosmetic: make syncing with 2.6 easier
+  o [libata] add ssleep() function
+  o [libata ahci] bump version to 1.00
+  o Add nth_page() helper
+  o Resync linux/ata.h with 2.6.x
+  o Remove silly comment from linux/ata.h
+  o [libata] remove dependence on PCI (2.4 stub version)
+  o [libata] bump versions, add MODULE_VERSION() tags
+  o [libata] fix DocBook bugs
+  o [libata ahci] minor fixes
+
+Jeremy Higdon:
+  o per-port LED control for sata_vsc
+
+John W. Linville:
+  o 3c59x: resync with 2.6
+
+Len Brown:
+  o [ACPI] BIOS workaround allowing devices to use reserved IO ports Author: David Shaohua Li http://bugzilla.kernel.org/show_bug.cgi?id=3049
+
+Manfred Spraul:
+  o Backport of the 0.30 forcedeth driver to 2.4. It's a new backport, starting from the 2.6 tree.
+
+Marcelo Tosatti:
+  o O.Sezer: cpqphp_pci.c size warning fix
+  o Cset exclude: m.c.p@kernel.linux-systeme.com|ChangeSet|20041122173550|59288
+  o Marc-Christian Petersen: VM documentation update
+  o Ignore vma's with PageReserved pages at get_user_pages()
+  o Andrea: get_user_pages handle ZERO_PAGE PG_reserved page, BUG otherwise
+  o Changed EXTRAVERSION to -pre1
+  o fix get_user_pages() change typo
+
+Mark Lord:
+  o Export ata_scsi_simulate() for use by non-libata drivers
+
+Matthijs Melchior:
+  o [libata ahci] fix rather serious (and/or embarassing) bugs
+
+Meelis Roos:
+  o ata.h undefined types in USB
+
+Mikael Pettersson:
+  o gcc34 fastcall mismatch fixes for rwsem-spinlock
+
+Mike Kravetz:
+  o Task name handling static copy v2.6 backport
+
+Nishanth Aravamudan:
+  o scsi/ahci: replace schedule_timeout() with msleep()/ssleep()
+
+Pete Zaitcev:
+  o USB: fix ohci_complete_add
+  o USB: ohci fix by Jes&Pete for Jessie
+
+Ralf Bächle:
+  o MIPS update
+  o MIPS: sound drivers for AMD Alchemy platforms
+  o MIPS: Configure.help updates
+  o MIPS: MAINTAINERS update
+  o MIPS documentation
+
+Randy Dunlap:
+  o oops on boot when initializing CDROM
+
+Solar Designer:
+  o Fix SCSI tape driver return code
+  o Fix 32-bit syscall emulation waste of CPU resources
+
+Tobias Lorenz:
+  o [libata sata_promise] s/sata/ata/
+
+Tony Battersby:
+  o sym53c8xx_2 error handler fix
+  o sym53c8xx_2 sniff inquiry fix
+  o sym53c8xx_2 Ultra 160 requires LVD
+  o make SCSI error handler preserve data transfer residual
+  o fix for scsi_unjam_host: Miscount of number of failed commands
+  o fix race condition in sg.c
+
+Özkan Sezer:
+  o ricoh.h, mem0 wrong definition v2.6 backport
+  o Wilfried Weissmann: hptraid v0.03 from -ac/redhat - minor fixes
+  o DAC960 firmware/alpha backport from 2.6
+  o Cure ISDN eicon size warning
+  o OPTI Viper-M/N+ chipset support (by Michael Mueller)
+  o hamradio scc warning type fix
+  o intermezzo, backport some fixes from 2.6
+  o intermezzo, backport some more fixes from 2.6
+  o intermezzo, fixes from cvs
+  o Mark InterMezzo as orphan
+  o ide-scsi update from ac/rh: Added transform for reading ATAPI tape drive block limits
+
